agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -376,3 +456,168 @@ func (cs *CredentialSpecResource) MarshallPlatformSpecificFields(credentialSpecR
 func (cs *CredentialSpecResource) UnmarshallPlatformSpecificFields(credentialSpecResourceJSON CredentialSpecResourceJSON) {
 	return
 }
+
+// This function makes a copy of the local credential spec file to avoid modifying customer resources
+// Example Usage:
+// copyCredentialSpecLocalFile("credentialspecdomainless:file://gmsa-cred.json") - single file
+// copyCredentialSpecLocalFile("credentialspecdomainless:file://gmsa-dir\gmsa-cred.json") - Relative path with Windows file path separators
+func (cs *CredentialSpecResource) copyCredentialSpecLocalFile(credentialspec string) (string, error) {
+	relativeFilePath := strings.TrimPrefix(credentialspec, "credentialspecdomainless:file://")
+	dir, originalFileName := filepath.Split(relativeFilePath)
+
+	// Generate unique filename
+	taskId, err := getTaskIDfromARN(cs.taskARN)
+	if err != nil {
+		return "", err
+	}
+	containerName := cs.credentialSpecContainerMap[credentialspec]
+	copiedFileName := fmt.Sprintf("%s_%s_%s", taskId, containerName, originalFileName)
+
+	// Copy new file in same directory
+	// Read all the contents of the original file
+	bytes, err := os.ReadFile(filepath.Join(cs.credentialSpecResourceLocation, dir, originalFileName))
+	if err != nil {
+		return "", err
+	}
+
+	// Copy all the contents to the destination file
+	err = os.WriteFile(filepath.Join(cs.credentialSpecResourceLocation, dir, copiedFileName), bytes, 0777)
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("file://%s", filepath.Join(dir, copiedFileName)), nil
+}
+
+func setTaskExecutionCredentialsRegKeys(taskCredentials credentials.IAMRoleCredentials, taskArn string) error {
+	if taskCredentials == (credentials.IAMRoleCredentials{}) {
+		err := errors.New("Unable to find execution role credentials while setting registry key")
+		return err
+	}
+
+	seelog.Errorf("SetTaskExecutionCredentialsRegKey")
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE, servicesRegistryKeyRoot, registry.WRITE)
+	if err != nil {
+		seelog.Errorf("Error opening %s key: %s", servicesRegistryKeyRoot, err)
+		return err
+	}
+	defer k.Close()
+
+	k2, _, err := registry.CreateKey(k, taskCredentialsRegistryKey, registry.WRITE)
```

Comment:
super nit/question: can we update k, k2 and  k3 names to be something meaningful like servicesRegistryKey, taskCredentialsRegistryKey and task correspondingly?

Reply:
yes will do when I publish final version with units